generator/booking_dataset.cpp

```diff
@@ -3,188 +3,31 @@
-{
-  if (dataPath.empty())
-    return;
-
-  ifstream dataSource(dataPath);
-  if (!dataSource.is_open())
-  {
-    LOG(LERROR, ("Error while opening", dataPath, ":", strerror(errno)));
-    return;
-  }
-
-  LoadHotels(dataSource, addressReferencePath);
-}
-
-BookingDataset::BookingDataset(istream & dataSource, string const & addressReferencePath)
-{
-  LoadHotels(dataSource, addressReferencePath);
-}
-
-size_t BookingDataset::GetMatchingHotelIndex(FeatureBuilder1 const & fb) const
-{
-  if (CanBeBooking(fb))
-    return MatchWithBooking(fb);
-  return kInvalidHotelIndex;
-}
-
-bool BookingDataset::CanBeBooking(FeatureBuilder1 const & fb) const
+bool BookingDataset::NecessaryMatchingConditionHolds(FeatureBuilder1 const & fb) const
 {
   if (fb.GetName(StringUtf8Multilang::kDefaultCode).empty())
     return false;
 
   return ftypes::IsHotelChecker::Instance()(fb.GetTypes());
 }
 
-BookingDataset::Hotel const & BookingDataset::GetHotelById(uint32_t const id) const
-{
-  auto const it = m_hotels.find(id);
-  CHECK(it != end(m_hotels), ("Got wrong hotel id:", id));
-  return it->second;
-}
-
-BookingDataset::Hotel & BookingDataset::GetHotelById(uint32_t const id)
-{
-  auto it = m_hotels.find(id);
-  CHECK(it != end(m_hotels), ("Got wrong hotel id:", id));
-  return it->second;
-}
-
-vector<uint32_t> BookingDataset::GetNearestHotels(ms::LatLon const & latLon, size_t const limit,
-                                                  double const maxDistance /* = 0.0 */) const
-{
-  namespace bgi = boost::geometry::index;
-
-  vector<uint32_t> indexes;
-  for_each(bgi::qbegin(m_rtree, bgi::nearest(TPoint(latLon.lat, latLon.lon), limit)),
-           bgi::qend(m_rtree), [this, &latLon, &indexes, maxDistance](TValue const & v)
-           {
-             auto const & hotel = GetHotelById(v.second);
-             double const dist = ms::DistanceOnEarth(latLon.lat, latLon.lon, hotel.lat, hotel.lon);
-             if (maxDistance != 0.0 && dist > maxDistance /* max distance in meters */)
-               return;
-
-             indexes.emplace_back(v.second);
-           });
-
-  return indexes;
-}
-
-void BookingDataset::BuildHotels(function<void(FeatureBuilder1 &)> const & fn) const
-{
-  for (auto const & item : m_hotels)
-    BuildHotel(item.second, fn);
-}
-
-void BookingDataset::BuildHotel(Hotel const & hotel,
-                                function<void(FeatureBuilder1 &)> const & fn) const
+void BookingDataset::BuildObject(SponsoredDataset::Object const & hotel,
+                                 function<void(FeatureBuilder1 &)> const & fn) const
 {
   FeatureBuilder1 fb;
   FeatureParams params;
 
   fb.SetCenter(MercatorBounds::FromLatLon(hotel.lat, hotel.lon));
 
   auto & metadata = params.GetMetadata();
-  metadata.Set(feature::Metadata::FMD_SPONSORED_ID, strings::to_string(hotel.id));
+  metadata.Set(feature::Metadata::FMD_SPONSORED_ID, strings::to_string(hotel.id.Get()));
```

Comment:
Not sure that this is a good idea to put all sponsored ids to a single metadata field.